### PR TITLE
librbd: cannot access non-primary image when mirroring force disabled 

### PR DESCRIPTION
**Pull request by dillaman**

### CODE REVIEW

**Comment by dillaman** [comment]
retest this please

File: src/test/librbd/mirror/test_mock_DisableRequest.cc [hunk 1]
```diff
@@ -0,0 +1,581 @@
...
+    mock_image_ctx, {
+      {"", encode(journal::ClientData{journal::ImageClientMeta{}})},
+      {"peer 1", encode(journal::ClientData{journal::MirrorPeerClientMeta{}})},
+      {"peer 2", encode(journal::ClientData{journal::MirrorPeerClientMeta{
+        "remote image id", {{"snap 1", boost::optional<uint64_t>(0)},
+                            {"snap 2", boost::optional<uint64_t>(0)}}}
+      })}
+    }, 0);
+  expect_journal_client_unregister(mock_image_ctx, "peer 1", -EINVAL);
+  expect_snap_remove(mock_image_ctx, "snap 1", 0);
```
Comment (trociny) [inline src/test/librbd/mirror/test_mock_DisableRequest.cc]: @dillaman Observing this test (also SnapRemoveError, and I suppose some other) fails sporadically (--gtest_repeat=100) due to a race between unregister_client from "peer 1" and snap_remove from "peer 2", as they run concurrently and snap_remove may finish earlier than client_unregister:
```
2016-10-30 12:46:15.266679 7ff8d07b1980 10 librbd::mirror::DisableRequest: 0x7ff8d926eef0 send_get_mirror_image
2016-10-30 12:46:15.266709 7ff8bd92d700 10 librbd::mirror::DisableRequest: 0x7ff8d926eef0 handle_get_mirror_image: r=0
2016-10-30 12:46:15.266712 7ff8bd92d700 10 librbd::mirror::DisableRequest: 0x7ff8d926eef0 send_get_tag_owner
2016-10-30 12:46:15.266736 7ff8b6ffd700 10 librbd::mirror::DisableRequest: 0x7ff8d926eef0 handle_get_tag_owner: r=0
2016-10-30 12:46:15.266738 7ff8b6ffd700 10 librbd::mirror::DisableRequest: 0x7ff8d926eef0 send_set_mirror_image
2016-10-30 12:46:15.266763 7ff8bd92d700 10 librbd::mirror::DisableRequest: 0x7ff8d926eef0 handle_set_mirror_image: r=0
2016-10-30 12:46:15.266764 7ff8bd92d700 10 librbd::mirror::DisableRequest: 0x7ff8d926eef0 send_notify_mirroring_watcher
2016-10-30 12:46:15.266771 7ff8b6ffd700 10 librbd::mirror::DisableRequest: 0x7ff8d926eef0 handle_notify_mirroring_watcher: r=0
2016-10-30 12:46:15.266773 7ff8b6ffd700 10 librbd::mirror::DisableRequest: 0x7ff8d926eef0 send_get_clients
2016-10-30 12:46:15.266813 7ff8bd92d700 10 librbd::mirror::DisableRequest: 0x7ff8d926eef0 handle_get_clients: r=0
2016-10-30 12:46:15.266817 7ff8bd92d700 10 librbd::mirror::DisableRequest: 0x7ff8d926eef0 send_unregister_client
2016-10-30 12:46:15.266826 7ff8bd92d700 10 librbd::mirror::DisableRequest: 0x7ff8d926eef0 send_remove_snap: client_id=peer 2, snap_name=snap 1
2016-10-30 12:46:15.266832 7ff8bd92d700 10 librbd::mirror::DisableRequest: 0x7ff8d926eef0 send_remove_snap: client_id=peer 2, snap_name=snap 2
2016-10-30 12:46:15.266870 7ff8bd92d700 10 librbd::mirror::DisableRequest: 0x7ff8d926eef0 handle_unregister_client: r=-22
2016-10-30 12:46:15.266873 7ff8bd92d700 -1 librbd::mirror::DisableRequest: failed to unregister remote journal client: (22) Invalid argument
```

**Comment by dillaman** [comment]
@trociny update pushed

**Comment by trociny** [comment]
lgtm

**Pull request merged**